### PR TITLE
Add xretry package with Retrier

### PR DESCRIPTION
**Pull request by robskillington**
cc @xichen2020 @mmihic @martin-mao @cw9

### CODE REVIEW

File: retry/types.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+// THE SOFTWARE.
+
+package xretry
+
+import (
+	"time"
+)
+
+// Fn is a function that can be retried
+type Fn func() error
```
Comment (xichen2020) [inline retry/types.go]: perhaps call this `RetryableFn`?

**Comment by xichen2020** [comment]
LGTM other than the nit

File: retry/types.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+// ContinueFn is a function that returns whether to continue attempting an operation
+type ContinueFn func(attempt int) bool
```
Comment (mmihic) [inline retry/types.go]: Probably need a way to indicate that you shouldn't retry on a particular error.

Comment (robskillington) [inline retry/types.go]: Addressed

**Comment by mmihic** [comment]
lgtm

**Pull request merged**